spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -97,18 +99,43 @@ object StandardizationJob {
     val dfAll: DataFrame = prepareDataFrame(schema, cmd, pathCfg.inputPath, dataset)
 
     try {
-      executeStandardization(performance, dfAll, schema, cmd, menasCredentials, pathCfg, recordIdGenerationStrategy)
+      val standardized = executeStandardization(performance, dfAll, schema, cmd, menasCredentials, pathCfg, recordIdGenerationStrategy)
+      standardized.persist()
```

Comment:
I was thinking of doing it a slightly different way. 
Instead of persisting the dataframe which is already saved to Parquet, we can read the output folder.
This will save time for saving data twice and might be even faster since the new dataframe will have an execution plan consisting of the singe operation - read.

Such an approach has an additional advantage. If something goes wrong the persisted dataframe will be recalculated and record ids might change. But if you read it from the output directory, there is no way record ids can change.

Reply:
Redone as suggested (no persisting, writing to parquet + reading back)

It is a little bit counter-intuitive, but you suggestion makes sense. Speed/efficiency-wise, I see this as a trade-off between:
 - write to disk once, read once while assuming that parquet r/w is efficient enough.
 - write to disk once, and persist (maybe write again to disk, maybe keep in memory - and we could actually set the persistence strategy IIRC).
(so not a clear-cut one being so much better than the other)

The point of having the data with the stable ids, that seems much more important. Would the this idea work as-is? I can see that if the data exists, the spark-job refuses to run, and on the other hand, if the std/conf process fails, the kafka-error-sending routine is not reached anyway - it's in the same try block.

Reply:
> The point of having the data with the stable ids, that seems much more important. Would the this idea work as-is?

I'm not 100% sure, can just speculate. I think it will work on the happy path (no executor failed). But if one of the executors fail and some data need to be recomputed, it might have different ids. That's my theory 😄